packages/test/test-gc-sweep-tests/src/test/gcDriverWrappers.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+import {
+    IDocumentService,
+    IDocumentServiceFactory,
+    IDocumentStorageService,
+    IResolvedUrl,
+    ISummaryContext,
+} from "@fluidframework/driver-definitions";
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+
+/**
+ * Wraps the given IDocumentStorageService to override the `uploadSummaryWithContext` method. It calls the
+ * `uploadSummaryCb` whenever a summary is uploaded by the client. The summary context can be updated in the
+ * callback before it is uploaded to the server.
+ */
+export function wrapDocumentStorageService(
+    innerDocStorageService: IDocumentStorageService,
+    uploadSummaryCb: (summaryTree: ISummaryTree, context: ISummaryContext) => ISummaryContext,
+) {
+    const outerDocStorageService = Object.create(innerDocStorageService) as IDocumentStorageService;
```

Comment:
Rather than cloning the whole IDocumentStorageService, maybe just copy the `uploadSummaryWithContext` function and monkey-patch over it with the wrapper function?

Reply:
I might just delete all of this

Reply:
This is probably not needed for this PR similar to the comment you made above